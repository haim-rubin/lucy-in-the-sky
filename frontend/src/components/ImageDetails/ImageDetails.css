
.image-details {margin: 1rem; position: relative;}
.image-details .extra-info{ display: flex;}
.image-details .description{
    visibility: hidden;
    position: absolute;
    top: 0;
    padding: .25rem;
    background: rgb(255,255,255);
    background-color: rgba(26, 26, 114, 0.3);
    border-radius: .2rem;
    margin: 1rem;
    color: whitesmoke;
}
.image-details .description.show{
    visibility:visible;
    transition: visibility 1s, opacity 0.5s linear;
}
.image-details .thumbnail img {
    border-radius: 1rem;
    width: 20rem;
    height: 20rem;
    object-fit: cover;
}
.favorite{padding: .1rem .5rem;}
.fa.fa-heart{color: #fff; font-size: 2rem; box-shadow: tomato;}
.fa.fa-heart.is-favorite {color:tomato;}
.image-details .extra-info {
    position: absolute;
    bottom: 2rem;
}

