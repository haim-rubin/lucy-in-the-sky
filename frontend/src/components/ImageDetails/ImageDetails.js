import React, { useState } from 'react'
import cx from 'classnames'

const ImageDetails = ({ description, url, isFavorite, setFavorite, imageId }) => {
    const [hover, setIsHover] = useState(false)
    return (
        <div className='image-details' onMouseEnter={() => {
            setIsHover(true)
        }} onMouseLeave={() => {
            setIsHover(false)
        }}>
            <div className='thumbnail'>
                <img src={url} alt={description} />
            </div>
            {
                <div className={cx('description', { 'show': hover } )}>
                {description}
                </div>
            }
            <div className='extra-info'>
                <div className='favorite' onClick={() => {
                    setFavorite({ imageId, isFavorite: !isFavorite})
                }}>
                    <i className={cx('fa fa-heart', { 'is-favorite': isFavorite } )}></i>
                </div>
            </div>
        </div>
    )
}


export default ImageDetails