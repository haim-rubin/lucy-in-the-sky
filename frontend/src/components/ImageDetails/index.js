import ImageDetails from './ImageDetails'
import './ImageDetails.css'

export default ImageDetails