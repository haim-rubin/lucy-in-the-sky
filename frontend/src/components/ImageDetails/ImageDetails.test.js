import React from 'react'
import renderer from 'react-test-renderer'

import ImageDetails from './ImageDetails'

it('renders correctly when there are no items', () => {
  const tree =
    renderer
        .create(
            <ImageDetails
                description="Hello"
                url="https://images.unsplash.com/photo-1494438639946-1ebd1d20bf85?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjEyNDU2M30"
                isFavorite={false}
                setFavorite ={() => {}} imageId="1" />
        )
        .toJSON()
  expect(tree).toMatchSnapshot()
})