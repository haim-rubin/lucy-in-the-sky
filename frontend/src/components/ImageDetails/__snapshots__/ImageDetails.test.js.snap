// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="image-details"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="thumbnail"
  >
    <img
      alt="Hello"
      src="https://images.unsplash.com/photo-1494438639946-1ebd1d20bf85?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjEyNDU2M30"
    />
  </div>
  <div
    className="description"
  >
    Hello
  </div>
  <div
    className="extra-info"
  >
    <div
      className="favorite"
      onClick={[Function]}
    >
      <i
        className="fa fa-heart"
      />
    </div>
  </div>
</div>
`;
